style_guides/pl_sql_comments_history.md

```diff
@@ -0,0 +1,51 @@
+# Comments and Change History
+
+When creating database objects or scripts you should always include a header comment and history. For packages, you should include this in both the package spec and body to track changes to each. 
+
+The header should contain the following information:
+
+* Description - a summary of the package and an enhancement number
+* List of authors
+* Link to source code repository
+
+We do not create or include version numbers for items in the change history. 
+
+Each time the object is changed a new line should be added to the header. Each history line should contain:

```

Comment:
I realise this is unpopular but I still feel file history should be looked at in the source control system not in the file itself.

Reply:
I like having an audit history at the top of object definitions and think it speeds up problem solving to have this displayed against the deployed object in the database.

Reply:
I get it, I just don't agree with it. If I want the history of a package/file I would look in source control. The history comments can be inconstant, incorrect, or just missing as it relies on a human remembering to do it. Git just has all the changes and who did them and when.

I know I am in the minority. I just wanted my opinion on the record 😄